mongorestore/mongorestore_test.go

```diff
@@ -1658,6 +1658,92 @@ func createTimeseries(dbName, coll string, client *mongo.Client) {
 	client.Database(dbName).RunCommand(context.Background(), createCmd)
 }
 
+func TestRestoreTimeseriesCollectionsWithMixedSchema(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+
+	ctx := context.Background()
+
+	sessionProvider, _, err := testutil.GetBareSessionProvider()
+	if err != nil {
+		t.Fatalf("No cluster available: %v", err)
+	}
+
+	defer sessionProvider.Close()
+
+	session, err := sessionProvider.GetSession()
+	if err != nil {
+		t.Fatalf("No client available")
+	}
+
+	fcv := testutil.GetFCV(session)
+	// The server fix for SERVER-84531 was only backported to 7.3.
+	if cmp, err := testutil.CompareFCV(fcv, "7.3"); cmp < 0 {
+		if err != nil {
+			t.Fatalf("Failed to get FCV: %v", err)
+		}
+		t.Skip("Requires server with FCV 7.3 or later")
+	}
+
+	dbName := "timeseries_test_DB"
+	collName := "timeseries_mixed_schema"
+	testdb := session.Database(dbName)
+	bucketColl := testdb.Collection("system.buckets." + collName)
+
+	testdb.Collection(collName).Drop(ctx)
+
+	restore, err := getRestoreWithArgs(ArchiveOption + "=testdata/timeseries_tests/mixed_schema_dump.archive")
+	require.NoError(t, err)
+
+	// Run mongorestore
+	result := restore.Restore()
+	require.NoError(t, result.Err)
+	require.Equal(t, int64(1), result.Successes)
+	require.Equal(t, int64(0), result.Failures)
+	defer testdb.Collection(collName).Drop(ctx)
+
+	count, err := testdb.Collection(collName).CountDocuments(ctx, bson.M{})
+	require.NoError(t, err)
+	require.Equal(t, int64(2), count)
+
+	count, err = bucketColl.CountDocuments(ctx, bson.M{})
+	require.NoError(t, err)
+	require.Equal(t, int64(1), count)
+
+	hasMixedSchema, err := timeseriesBucketsMayHaveMixedSchemaData(bucketColl)
+	require.NoError(t, err)
+	require.False(t, hasMixedSchema)
```

Comment:
I'm a bit confused why this is succeeding- my understanding of the server change is that we need to set hasMixedSchema=true in order to perform these direct writes?